Tests/tests/test_ActivityMonitor.cpp

```diff
@@ -0,0 +1,219 @@
+/**
+ * If not stated otherwise in this file or this component's LICENSE
+ * file the following copyright and licenses apply:
+ *
+ * Copyright 2022 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+#include "gtest/gtest.h"
+
+#include "ActivityMonitor.h"
+
+#include "FactoriesImplementation.h"
+#include "ServiceMock.h"
+#include "IarmBusMock.h"
+#include <fstream>
+#include <unistd.h>
+
+using namespace WPEFramework;
+
+namespace {
+const string regFile = _T("/opt/waylandregistry.conf");
+const uint8_t regFileData[] = "{\"waylandapps\":[{\"name\" : \"memcheck-amd64-\",\"binary\" : \"/usr/lib/x86_64-linux-gnu/valgrind/memcheck-amd64-%\"}]}";
+}
+
+class ActivityMonitorTest : public ::testing::Test {
+protected:
+    Core::ProxyType<Plugin::ActivityMonitor> plugin;
+    Core::JSONRPC::Handler& handler;
+    Core::JSONRPC::Connection connection;
+    string response;
+
+    ActivityMonitorTest()
+        : plugin(Core::ProxyType<Plugin::ActivityMonitor>::Create())
+        , handler(*(plugin))
+        , connection(1, 0)
+    {
+        EXPECT_EQ(string(""), plugin->Initialize(nullptr));
+    }
+    virtual ~ActivityMonitorTest() override
+    {
+        IarmBus::getInstance().impl = nullptr;
+        plugin->Deinitialize(nullptr);
+    }
+};
+
+TEST_F(ActivityMonitorTest, registeredMethods)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("enableMonitoring")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("disableMonitoring")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getApplicationMemoryUsage")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getAllMemoryUsage")));
+}
+
+TEST_F(ActivityMonitorTest, disableMonitoring)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("disableMonitoring"), _T("{}"), response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+}
+
+TEST_F(ActivityMonitorTest, enableMonitoringEmptyConfig)
+{
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("enableMonitoring"), _T("{\"config\":[],\"memoryIntervalSeconds\":\"0\", \"cpuIntervalSeconds\":\"0\"}"), response));
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("enableMonitoring"), _T("{\"config\":[{\"appPid\": 6763, \"memoryThresholdMB\": 10,\"cpuThresholdPercent\": 50,\"cpuThresholdSeconds\": 2}],\"memoryIntervalSeconds\":\"0\", \"cpuIntervalSeconds\":\"0\"}"), response));
+}
```

Comment:
where do these magical numbers come from?

Reply:
These can be any random numbers. The test case is "enableMonitoring" should return "ERROR_GENERAL", with a valid parameters, since the app registry file is not available.